sklearn/preprocessing/_encoders.py

```diff
@@ -350,6 +363,27 @@ def _compute_drop_idx(self):
                                             len(self.drop)))
             missing_drops = [(i, val) for i, val in enumerate(self.drop)
                              if val not in self.categories_[i]]
+
+            missing_drops = []
+            drop_idxs = []
+            for i, (val, cat_list) in enumerate(zip(self.drop,
+                                                    self.categories_)):
+                if not is_scalar_nan(val):
```

Comment:
So this is for the user explicitly specifying that nan should be dropped? Because 'first' will never result in nan being dropped, right?